src/main/scala/Task2/package.scala

```diff
@@ -0,0 +1,24 @@
+import scala.annotation.tailrec
+import scala.concurrent.duration.{FiniteDuration, _}
+
+package object Task2 {
+
+  @tailrec
+  final def retryFuture[A](block: () => A,
+                           acceptResult: A => Boolean,
+                           retries: List[FiniteDuration]): A = {
+    import scala.concurrent._
+    import ExecutionContext.Implicits.global
+    val future: Future[A] = Future {
+      block()
+    }
+    val result = Await.result(future, 1.seconds)
```

Comment:
no Await should be used at all here, they are blocking threads, and we told in the class why it's bad.